src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +10,27 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationValidationException("User cannot be null");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationValidationException("Login must contain at least 6 characters");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new
+                    RegistrationValidationException("Password must contain at least 6 characters");
```

Comment:
Similar to the login, the password length check should also include a null check before attempting to access `.length()` to avoid a NullPointerException.